seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumJSEnv.scala

```diff
@@ -67,8 +67,42 @@ object SeleniumJSEnv {
         materializer = DefaultFileMaterializer,
         driverFactory = Config.defaultFactory)
 
-    def withMaterializer(materializer: FileMaterializer): Config =
-      copy(materializer = materializer)
+    /** Materializes purely virtual files into a temp directory.
+     *
+     *  Materialization is necessary so that virtual files can be referred to by
+     *  name. If you do not know/care how your files are referred to, this is a
+     *  good default choice. It is also the default of [[SeleniumJSEnv.Config]].
+     */
+    def withMaterializeInTemp: Config =
+      copy(materializer = DefaultFileMaterializer)
+
+    /** Materializes files in a static directory of a user configured server.
+     *
+     *  This can be used to bypass cross origin access policies.
+     *
+     *  @param contentDir Static content directory of the server. The files will
+     *      be put here. Will get created if it doesn't exist.
+     *  @param webRoot URL making `contentDir` accessible thorugh the server.
+     *      This must have a trailing slash to be interpreted as a directory.
+     *
+     *  @example
+     *
+     *  The following will make the browser fetch files using the http:// schema
+     *  instead of the file:// schema. The example assumes a local webserver is
+     *  running and serving the ".tmp" directory at http://localhost:8080.
+     *
+     *  <pre>
+     *    jsSettings(
+     *      jsEnv := new SeleniumJSEnv(
+     *          org.openqa.selenium.remote.DesiredCapabilities.firefox(),
+     *          SeleniumJSEnv.Config()
+     *            .withMaterializeInServer(".tmp", "http://localhost:8080/")
+     *      )
+     *    )
+     *  </pre>
```

Comment:
Scaladoc has dedicated syntax for code snippets: `{{{`..`}}}`-delimited blocks (usually without further increasing indentation inside them).